### PR TITLE
API calls to fetch and update the Automate Workspace

### PR DESCRIPTION
**Pull request by mkanoor**
Added API calls to fetch and update the Automate Workspace.
Each Automate Workspace has a unique GUID and they can be only retrieved using the GUID.

### CODE REVIEW

**Comment by mkanoor** [comment]
Depends on https://github.com/ManageIQ/manageiq/pull/15817 and
https://github.com/ManageIQ/manageiq-schema/pull/50

File: app/controllers/api/automate_workspaces_controller.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      end
+      workspace = data["workspace"]
+      state_var = data["state_var"]
+      if workspace.blank? && state_var.blank?
+        raise BadRequestError, "No workspace or state_var specified for edit"
+      end
+      current_output = obj.output || {}
+      obj.output = current_output.deep_merge(data)
+      obj.save
+      obj
```
Comment (Fryguy) [inline app/controllers/api/automate_workspaces_controller.rb]: None of this belongs in a controller.  The api controllers should be stupid, and the model should have all of this "how to patch an atuomate workspace" knowledge.

Comment (abellotti) [reply app/controllers/api/automate_workspaces_controller.rb]: 👍 model/business logic stays in model.

Comment (imtayadeway) [reply app/controllers/api/automate_workspaces_controller.rb]: > The api controllers should be stupid

❤️ ❤️ ❤️

File: app/controllers/api/automate_workspaces_controller.rb [hunk 2]
```diff
@@ -18,7 +18,7 @@ def edit_resource(_type, id, data = {})
       if obj.nil?
         raise NotFoundError, "Invalid Workspace #{id} specified"
       end
-      obj.update_output(data)
+      obj.output = data
```
Comment (abellotti) [inline app/controllers/api/automate_workspaces_controller.rb]: do we need a begin/rescue here ?

**Comment by imtayadeway** [comment]
Hi! It looks like your PR has been affected by the recently merged https://github.com/ManageIQ/manageiq-api/pull/40, and as a result will need to be rebased. First, I apologize for any inconvenience caused. After rebasing, you'll need to update some specs in order for them to pass. In particular you'll need to change any path helpers, applying the following pattern:

```
vms_url => api_vms_url
vms_url(vm.id) => api_vm_url(nil, vm) 
vms_url(vm.compressed_id) => api_vm_url(nil, vm.compressed_id)
"#{vms_url(vm.id)}/tags" => api_vm_tags_url(nil, vm)
"#{vms_url(vm.compressed_id}/tags" => api_vm_tags_url(nil, vm.compressed_id)
"#{vms_url(vm.id)}/tags/#{tag.id}" => api_vm_tag_url(nil, vm, tag)
"#{vms_url(vm.compressed_id)}/tags/#{tag.compressed_id}" => api_vm_tag_url(nil, vm.compressed_id, tag.compressed_id)
```

If you run into any issues, please ping me and I will try to help you out.

Many thanks!
Tim

**Comment by abellotti** [comment]
LGTM!!  We'll update this once we have the update in the base to support arbitrary resource identifier (i.e. guid instead of id).

**Pull request merged**